plugins/module_utils/consul.py

```diff
@@ -232,6 +232,12 @@ def prepare_object(self, existing, obj):
             existing[k] = v
         return existing
 
+    def get_first_appearing_identifier(self, identifiers):
```

Comment:
What about moving `unique_identifier` in the base class to a `unique_identifiers` list, then you'd just have to set this once and it can be used consistently. (Will then need changes through the other modules as well. Maybe we do that in an extra PR first?)

Reply:
Yes, I had the same idea. My new push may be better. 
Or we can do it like you said.